video/out/wayland_common.c

```diff
@@ -3467,8 +3467,8 @@ static void set_color_management(struct vo_wayland_state *wl)
     wp_image_description_creator_params_v1_set_primaries_named(image_creator_params, primaries);
     wp_image_description_creator_params_v1_set_tf_named(image_creator_params, transfer);
 
-    pl_color_space_infer(&wl->target_params.color);
-    struct pl_hdr_metadata hdr = wl->target_params.color.hdr;
+    pl_color_space_infer(&color);
```

Comment:
`wl->target_params` should be kept as close to output as possible. What is the problem with updating unknown parameters in it?

Reply:
You're right, this just happens to result in a fix but isn't actually the right fix.

The problem is that the file in the issue has `min_luma == 0`, so when we call `pl_color_space_infer` on it, it sets `min_luma = 0.000001` which produces completely incorrect output

Reply:
wlshm was deliberately omitted from color management since it's currently hardcoded to only output BGR0 or 0RGB. Had some vague plans to add more shm formats but I recall I got some weird behavior on compositors and shelved it.

Reply:
Sure it could still work but I don't think there's any value is there?